core/src/main/scala/com/rallyhealth/vapors/factfilter/evaluator/InterpretExprAsFunction.scala

```diff
@@ -213,7 +224,7 @@ final class InterpretExprAsFunction[F[_] : Foldable, V, P]
   }
 
   override def visitReturnInput(expr: Expr.ReturnInput[F, V, P]): Input[F, V] => ExprResult[F, V, F[V], P] = { input =>
-    resultOfPureExpr(expr, input, input.value, input.evidence) {
+    resultOfPureExpr(expr, input, input.value, input.evidence ++ Evidence.fromAnyOrNone(input.value)) {
```

Comment:
I should document this. Basically, if the input is a Fact or an Iterable[Fact], then it is added to its own evidence when moved to the output value.

Reply:
Yes, please do!

Reply:
I think I'm gonna wait until after I rework the evidence tracking to document it, since it is fairly broken right now.